extensions/amp-ad/0.1/amp-ad-api-handler.js

```diff
@@ -121,21 +122,36 @@ export class AmpAdApiHandler {
       // creative as it will not expect having to send the render-start message.
       this.iframe_.style.visibility = 'hidden';
     }
-    listenForOnce(this.iframe_, 'render-start', () => {
-      if (!this.iframe_) {
-        return;
+
+    const promise = new Promise(resolve => {
+      listenForOnce(this.iframe_, 'render-start', () => {
+        resolve();
+      }, this.is3p_);
+    });
+
+    this.renderStartPromise_ = timerFor(this.baseInstance_.win).timeoutPromise(
+        2000, promise,
+        'fail to receive render-start event from ad server, timeout');
+
+    this.renderStartPromise_.then(() => {
+      //TODO: report performance
+      if (this.iframe_) {
+        this.iframe_.style.visibility = '';
       }
-      if (this.baseInstance_.renderStartResolve_) {
-        this.baseInstance_.renderStartResolve_();
-        this.baseInstance_.renderStartResolve_ = null;
+    }).catch(() => {
+      //TODO: report performance
+      if (this.iframe_) {
+        this.iframe_.style.visibility = '';
```

Comment:
It does. And when you `catch` it, you actually kill the error. So we need to make sure it gets reported one way or another.